internal/api/controllers/private/runsCreate.go

```diff
@@ -7,13 +7,68 @@ import (
 	"playbook-dispatcher/internal/api/middleware"
 	"playbook-dispatcher/internal/common/db"
 	dbModel "playbook-dispatcher/internal/common/model/db"
+	"playbook-dispatcher/internal/common/satellite"
 	"playbook-dispatcher/internal/common/utils"
 
 	"github.com/google/uuid"
 	"github.com/labstack/echo/v4"
 	"github.com/spf13/viper"
 )
 
+func CheckReqFields(runInput RunInput) (ansible bool, satellite bool) {
+	ansible, satellite = true, true
+
+	if runInput.PlaybookName == nil || runInput.PlaybookRunUrl == nil {
+		satellite = false
+	}
+	if runInput.Initiator == nil {
+		satellite = false
+	}
+	if runInput.ExecutorConfigs == nil {
+		satellite = false
+	}
+	if runInput.ExecutorConfigs != nil && (runInput.ExecutorConfigs.SatId == nil || runInput.ExecutorConfigs.SatOrgId == nil) {
+		satellite = false
+	}
+
+	if runInput.Hosts == nil {
+		satellite = false
+		ansible = false
```

Comment:
The hosts field is optional for the plain rhc type of request.

Reply:
changed!